Readme update (#180)
ministryofjustice/court-case-service: README.md

# Court Case Service
### Service to access court cases imported from HMCTS Libra court lists

---

## Setup your machine

> In the root folder of your machine start by checking if HomeBrew is installed on your computer with the command : `brew -version` 

Reviewer: The bit about Java 11 is definitely useful but the rest might be a bit more detail than we'd normally go into. It's useful stuff but different users might have different ways of installing dependencies (for example Windows/Linux users) and might encounter different problems so trying to document for every possibility quickly becomes intractable!  

Happy to leave it in as it as we don't have huge amounts of documentation, you're right we've been a bit lax on that so I'm very glad you're taking the time to make improvements!

Author: How about making a section for Mac and leave an open door for people Window/Linux? Maybe the next after me will be happy to fill these section :) ?

Reviewer: I'd leave it as it is, we only use macs anyway so it was mostly to illustrate a point, if some intrepid Windows user does come along and then they can of course submit a PR!

#### 5. Reopen a new window and run :   
~5.1.~ `$ /usr/libexec/java_home -V` 
~5.2.~ `$ jenv add <choose one/all the versions you have found when you ran the previous command>`
*E.g.: `jenv add /Library/Java/JavaVirtualMachines/openjdk-11.0.2.jdk/Contents/Home`*

## Dependencies

The service has an attached Postgres database as well as several back-ends. The details of which can be found in the `docker-compose.yml` file.
:::info
⚠️ To open all of these, complete the Building and Running section.
:::

**Default port**
Starts the application on port '8080'. To override, set server.port (eg SERVER_PORT=8099 java -jar etc etc)

**Swagger UI**
The project builds swagger specifications which can be examined and tested via the Swagger UI. http://localhost:8080/swagger-ui.html

**Application health** : `$ curl -X GET http://localhost:8080/health`
**Application info** : `$ curl -X GET http://localhost:8080/info`
**Application Ping** : `$ curl -X GET http://localhost:8080/ping`
**Application Feature Flags** : `$ curl -X GET http://localhost:8080/feature-flags`
**To Check dependency versions** : `$ ./gradlew dependencyUpdates`

#### *Flyway commands*

> **Migrate database** : `$ gradle flywayMigrate -i`

Reviewer: ```suggestion
**Migrate database** : `$ ./gradlew flywayMigrate -i`
```

> **View details and status information about all migrations** : `$ gradle flywayInfo`

Reviewer: ```suggestion
**View details and status information about all migrations** : `$ ./gradlew flywayInfo`
```

> **Baseline an existing database, excluding all migrations up to and including baselineVersion** :  `$ gradle flywayBaseline`

Reviewer: Because Spring Boot automatically bootstraps our database we don't generally use the Gradle commands much (though it's still useful to have) - the one exception to this is `flywayClean` which cleans away all migrations applied to a database and removes the flyway schema. This can be useful if you somehow manage to corrupt your migrations, which I've managed a fair few times 😄 . 

Worth noting that we usually use the provided `./gradlew` (Gradle wrapper) file rather than `gradle`. This ensures you've always got the right version of gradle for a given repo but is otherwise identical in usage.

```suggestion
**Baseline an existing database, excluding all migrations up to and including baselineVersion** :  `$ ./gradlew flywayBaseline`
**Clean schema** : `$ ./gradlew flywayClean`
```
